### PR TITLE
Moving Butter Knife and Renderers libraries from Rosie to the sample project, updated sample butterknife version 7.0.1 -> 8.0.1

### LINKED ISSUE #70: Remove ButterKnife and Renderers library.
In order to reduce the number of dependencies of this framework I suggest to remove ButterKnife and Renderers libraries. @Serchinastico, do you agree? @luis-ibanez instead of update ButterKnife we should delete it and keep the code related to this library and the renderers library just in the example project.

### PR DESCRIPTION
**Pull request by luis-ibanez**
There was a warning for using a @Nullable over a @OnClick method in MarvelActivity.java, I have updated Butterknife, replace that @Nullable for @Optional and updated all the @Bind annotation for @BindView. Fix #70

### CODE REVIEW

**Comment by pedrovgs** [comment]
Thank you so much  @luis-ibanez 👍

**Comment by Serchinastico** [comment]
Awesome PR :+1:

**Comment by luis-ibanez** [comment]
I'm researching why the tests failed, it seems there is a problem finding the loading view inside the include. Will commit when fixed.

**Comment by Serchinastico** [comment]
Yup, we got to the same point. Butterknife doesn't bind the views correctly. Thanks for taking a look, we will wait for your update!

**Comment by luis-ibanez** [comment]
Fixes #70

File: README.md [hunk 1]
```diff
@@ -418,6 +404,12 @@ or to your ``pom.xml`` if you are using Maven
 
 ```
 
+Changelog
+-----------------
+
+We have removed the direct support to Renderers and Butter Knife libraries on Rosie to simplify its use and make easier its support. The sample is still using this libraries, so you can find an example of how to integrate this libraries in your project.
+
+
```
Comment (pedrovgs) [inline README.md]: @luis-ibanez  we are not using a changelog for this project we use the release information to notify the release changes if needed.

File: README.md [hunk 2]
```diff
@@ -172,9 +161,6 @@ To understand when the lifecycle methods are called take a look at the following
 | ``pause``       | ``onPause``    | ``onPause``        |
 | ``destroy``     | ``onDestroy``  | ``onDestroy``      |
 
-####Renderers
-
-Finally, Rosie includes the [Renderers] [renderers] library to simplify your ``RecyclerView`` handling code. If you decide to use Renderers, remember to extend directly from ``RosieRenderer<T>`` to have ButterKnife injections for free in your renderer views.
```
Comment (pedrovgs) [inline README.md]: Remember to remove the links related to ``Renderers`` and ``ButterKnife`` at the end of the file.

File: sample/src/main/java/com/karumi/rosie/sample/base/view/activity/MarvelActivity.java [hunk 3]
```diff
@@ -16,20 +16,29 @@
...
-  @Nullable @OnClick(R.id.iv_toolbar_back) public void onBackButtonClicked() {
+  @Optional @OnClick(R.id.iv_toolbar_back) public void onBackButtonClicked() {
     finish();
   }
 
+  @Override
+  protected void onPreparePresenter() {
+    super.onPreparePresenter();
+    ButterKnife.bind(this);
+  }
```
Comment (pedrovgs) [inline sample/src/main/java/com/karumi/rosie/sample/base/view/activity/MarvelActivity.java]: We should create another method invoked just after the setContentView method. Something like ``onPrepareActivity`` or just override ``setContentView`` method to add the butter knife call. Thoughts @Serchinastico?

File: sample/src/main/java/com/karumi/rosie/sample/base/view/fragment/MarvelFragment.java [hunk 4]
```diff
@@ -20,7 +20,16 @@
...
 import com.karumi.rosie.view.RosieFragment;
 
+import butterknife.ButterKnife;
+
 public abstract class MarvelFragment extends RosieFragment {
+
+  @Override
+  protected void onPreparePresenter() {
+    super.onPreparePresenter();
+    ButterKnife.bind(this, getView());
```
Comment (pedrovgs) [inline sample/src/main/java/com/karumi/rosie/sample/base/view/fragment/MarvelFragment.java]: In the case of the fragment create a new method is not needed but this call to ButterKnife should be moved to ``onCreateView`` method.

**Comment by pedrovgs** [comment]
@luis-ibanez you have some checkstyle violations:

```
/home/travis/build/Karumi/Rosie/sample/src/main/java/com/karumi/rosie/sample/base/view/fragment/MarvelFragment.java:20:8: Unused import - android.support.annotation.Nullable.
```

**Comment by luis-ibanez** [comment]
@pedrovgs the build failed again

**Comment by pedrovgs** [comment]
Sorry, I'm restarting it again..the problem is the Android Emulator and the Travis-CI environments :S

**Comment by pedrovgs** [comment]
I've no idea about what's going on with this build :S We always get a timeout during the dexing stage... :_(

**Comment by luis-ibanez** [comment]
I close this PR just to response better to the #70 issue, and make the update in a separate PR given the problems making everything together is causing.

**Comment by pedrovgs** [comment]
ok

**Pull request closed without merging**